### ISSUE TITLE
Optional dependencies get always installed

### ISSUE THREAD

**Issue by jonschill24** [mindee/doctr#1658]
### Bug description

Although I did not specify to install the [html] verision of doctr, weasyprint gets installed as a dependency of doctr.
This does happen for other optional dependencies as well. For example matplotlib.
I did not expect these packages to be installed and I would prefer to keep external dependencies to the necessary minimum.

### Code snippet to reproduce the bug

My requirements.txt contains:
```
python-doctr[torch]~=0.8.1
```

I install via
```
pip install -r requirements.txt --verbose
```

### Error traceback

The output of pip install contains:
```
Collecting weasyprint>=55.0 (from python-doctr~=0.8.1->-r requirements.txt (line 17))
```

### Environment

DocTR version: v0.8.1
TensorFlow version: N/A
PyTorch version: 2.2.2 (torchvision 0.17.2)
OpenCV version: 4.9.0
OS: Mac OSX 14.5
Python version: 3.11.7
Is CUDA available (TensorFlow): N/A
Is CUDA available (PyTorch): No
CUDA runtime version: Could not collect
GPU models and configuration: Could not collect
Nvidia driver version: Could not collect
cuDNN version: Could not collect

### Deep Learning backend

is_tf_available: False
is_torch_available: True

**Comment by felixdittrich92** [comment]
Hi @jonschill24 👋,

That's only possible with the main branch atm and will be available with the next release (v0.9.0) soon. 

Best regards,
Felix

**Comment by jonschill24** [comment]
Alright, thanks a lot for the quick reply!
